code/game/turfs/simulated/floor/asteroid.dm

```diff
@@ -0,0 +1,322 @@
+	//note that this proc does not call ..()
+	if(!W || !user)
+		return 0
+
+	if((istype(W, /obj/item/shovel) || istype(W, /obj/item/pickaxe)))
+		var/turf/T = get_turf(user)
+		if(!istype(T))
+			return
+
+		if(dug)
+			to_chat(user, "<span class='warning'>This area has already been dug!</span>")
+			return
+
+		to_chat(user, "<span class='notice'>You start digging...</span>")
+		playsound(src, W.usesound, 50, 1)
+		if(do_after(user, 20 * W.toolspeed, target = src))
+			to_chat(user, "<span class='notice'>You dig a hole.</span>")
+			gets_dug()
+			return
+
+	if(istype(W,/obj/item/storage/bag/ore))
+		var/obj/item/storage/bag/ore/S = W
+		if(S.collection_mode == 1)
+			for(var/obj/item/stack/ore/O in src.contents)
+				O.attackby(W,user)
+				return
+
+	if(istype(W, /obj/item/stack/tile))
+		var/obj/item/stack/tile/Z = W
+		if(!Z.use(1))
+			return
+		ChangeTurf(Z.turf_type)
+		playsound(src, 'sound/weapons/Genhit.ogg', 50, 1)
+
+/turf/simulated/floor/plating/asteroid/gets_drilled()
+	if(!dug)
+		gets_dug()
+	else
+		..()
+
+/turf/simulated/floor/plating/asteroid/proc/gets_dug()
+	if(dug)
+		return
+	for(var/i in 1 to 5)
+		new sand_type(src)
+	dug = 1
+	icon_plating = "[environment_type]_dug"
+	icon_state = "[environment_type]_dug"
+	slowdown = 0
+	return
+
+/turf/simulated/floor/plating/asteroid/singularity_act()
+	return
+
+/turf/simulated/floor/plating/asteroid/singularity_pull(S, current_size)
+	return
+
+/turf/simulated/floor/plating/asteroid/basalt
+	name = "volcanic floor"
+	baseturf = /turf/simulated/floor/plating/asteroid/basalt
+	icon_state = "basalt"
+	icon_plating = "basalt"
+	environment_type = "basalt"
+	sand_type = /obj/item/stack/ore/glass/basalt
+	floor_variance = 15
+
+/turf/simulated/floor/plating/asteroid/basalt/lava //lava underneath
+	baseturf = /turf/simulated/floor/plating/lava/smooth
+
+/turf/simulated/floor/plating/asteroid/basalt/airless
+	temperature = TCMB
+	oxygen = 0
+	nitrogen = 0
+
+/turf/simulated/floor/plating/asteroid/basalt/Initialize()
+	. = ..()
+	set_basalt_light(src)
+
+/proc/set_basalt_light(turf/simulated/floor/B)
+	switch(B.icon_state)
+		if("basalt1", "basalt2", "basalt3")
+			B.set_light(2, 0.6, LIGHT_COLOR_LAVA) //more light
+		if("basalt5", "basalt9")
+			B.set_light(1.4, 0.6, LIGHT_COLOR_LAVA) //barely anything!
+
+/turf/simulated/floor/plating/asteroid/basalt/gets_dug()
+	if(!dug)
+		set_light(0)
+	..()
+
+///////Surface. The surface is warm, but survivable without a suit. Internals are required. The floors break to chasms, which drop you into the underground.
+
+/turf/simulated/floor/plating/asteroid/basalt/lava_land_surface
+	oxygen = 14
+	nitrogen = 23
+	temperature = 300
+	baseturf = /turf/simulated/floor/plating/lava/smooth/lava_land_surface
+
+/turf/simulated/floor/plating/asteroid/airless
+	temperature = TCMB
+	oxygen = 0
+	nitrogen = 0
+	turf_type = /turf/simulated/floor/plating/asteroid/airless
+
+#define SPAWN_MEGAFAUNA "MEGAFAUNA"
```

Comment:
TG handles this so weirdly. I dunno why they did what they did.

Reply:
Something to do with limiting the bubblegum spawn to 1, but that didn't fully work on our end so I had to turn it into a global list.